### PR TITLE
Add conventional npm scripts

### PR DESCRIPTION
**Pull request by ChristophWurst**
This is a draft of what @skjnldsv asked me for in regards to https://github.com/nextcloud/documentation/issues/967.

This is a quite opinionated view on how npm stuff should looks like, so feedback/input is highly appreciated :v:

### CODE REVIEW

File: developer_manual/app/npm.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+This command should be added to ``package.json`` as ``dev`` script:
+
+.. code-block:: json
+
+	{
+	  "name": "myapp",
+	  "scripts": {
+		"build": "webpack --progress --config webpack.prod.js",
+        "dev": "webpack --progress --config webpack.dev.js"
```
Comment (skjnldsv) [inline developer_manual/app/npm.rst]: we're all using watch I think, shall we recommend that?

File: developer_manual/app/npm.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
+===
+NPM
+===
+
+The node package manager ``npm`` is a widely used tool to manage JavaScript dependencies. In
+order to make the project setup easier for everybody, we recommend to follow the convention
+of typical scripts like ``build``, ``dev`` and ``test`` in the ``package.json`` file.
```
Comment (skjnldsv) [inline developer_manual/app/npm.rst]: shall we also add `lint`and `stylelint` ?

**Comment by skjnldsv** [comment]
Looks really nice!! I added a few comments :v:

File: developer_manual/app/npm.rst (hunk 1 above)
Comment (ChristophWurst) [reply developer_manual/app/npm.rst]: yeah, unsure about that. watch is a dev build that watches (surprise!) changed files. why is there a `dev` build as well? what's that used for?

File: developer_manual/app/npm.rst (hunk 2 above)
Comment (ChristophWurst) [reply developer_manual/app/npm.rst]: yep, good idea :+1:

File: developer_manual/app/npm.rst (hunk 1 above)
Comment (ChristophWurst) [reply developer_manual/app/npm.rst]: have you ever used the `dev` script?

Comment (juliushaertl) [reply developer_manual/app/npm.rst]: I think just having the watch script is fine. If we add watch we should add a note that it probably requires increasing `fs.inotify.max_user_watches` which was an issue @violoncelloch had when trying to watch.

Comment (skjnldsv) [reply developer_manual/app/npm.rst]: > have you ever used the `dev` script?

Rarely, but yes when I need a single build and then do another command (and that I'm too lazy to open a new shell :grin: )

**Comment by ChristophWurst** [comment]
> [ChristophWurst](/ChristophWurst) added this to **IN PROGRESS (max 3 PRs)** in [Christoph's Tasks](/orgs/nextcloud/projects/4#card-15295174) via automation [11 days ago](#event-1995072911)

Haven't forgotten but got sidetracked. Will continue/finish soon :v:

File: developer_manual/app/npm.rst (hunk 2 above)
Comment (skjnldsv) [reply developer_manual/app/npm.rst]: Because this is separated behaviours I guess? 
I don't have such strong preference :)

Comment (ChristophWurst) [reply developer_manual/app/npm.rst]: Me neither. Squash & merge?

Comment (skjnldsv) [reply developer_manual/app/npm.rst]: LETS GO! :muscle:

File: developer_manual/app/npm.rst [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    }
+  }
+
+If style linting is a separate script, ``stylelint`` shall be used as conventional script name:
+
+.. code-block:: json
+
+  {
+    "scripts": {
+      "stylelint": "stylelint src"
```
Comment (skjnldsv) [inline developer_manual/app/npm.rst]: You can also check the css folder?

Comment (skjnldsv) [inline developer_manual/app/npm.rst]: You can also check the css folder?

File: developer_manual/app/npm.rst [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+the bundler to listen to file changes and (incrementally) rebuild the project.
+
+This command should be added to ``package.json`` as ``watch`` script:
+
+.. code-block:: json
+
+  {
+    "name": "myapp",
+    "scripts": {
+    "build": "webpack --progress --config webpack.prod.js",
```
Comment (skjnldsv) [inline developer_manual/app/npm.rst]: Indentation

File: developer_manual/app/npm.rst (hunk 3 above)
Comment (ChristophWurst) [reply developer_manual/app/npm.rst]: what do you mean? The script is just an example. The essential part here is the script name.

Comment (skjnldsv) [reply developer_manual/app/npm.rst]: I guess :p

**Comment by ChristophWurst** [comment]
> Rebase to master and go :)

Done :dancer:

**Pull request merged**